common/api.go

```diff
@@ -6,6 +6,10 @@ import (
 )
 
 // JSHandleAPI is the interface of an in-page JS object.
+//
+// TODO: Find a way to move this to a concrete type. It's too difficult to
+// do that right now because of the tests and the way we're using the
+// JSHandleAPI interface.
```

Comment:
If the complexity here is because of tests, could we define specific `JSHandle` interfaces in the tests and in the rest use the concrete type? So at least we can get rid of this `common/api.go` file? It's a pity to leave just one type as interface everywhere.

Reply:
`JSHandle` complexity is dispersed over the whole module. I've tried to move this to a concrete type but couldn't. Not only because of the tests but also because of the code dependencies. We can discuss how to refactor this and send another PR, as it's quite complex to do in this PR.